### ISSUE TITLE
S3 headbucket command is triggering W3037

### ISSUE THREAD

**Issue by nick-disney** [aws-cloudformation/cfn-lint#4113]
### CloudFormation Lint Version

1.34.0

### What operating system are you using?

Mac and Docker

### Describe the bug

We are receiving the following warning on our template:

```
W3037 'headbucket' is not one of ['abortmultipartupload', 'associateaccessgrantsidentitycenter', 'bypassgovernanceretention', 'createaccessgrant', 'createaccessgrantsinstance', 'createaccessgrantslocation', 'createaccesspoint', 'createaccesspointforobjectlambda', 'createbucket', 'createbucketmetadatatableconfiguration', 'createjob', 'createmultiregionaccesspoint', 'createstoragelensgroup', 'deleteaccessgrant', 'deleteaccessgrantsinstance', 'deleteaccessgrantsinstanceresourcepolicy', 'deleteaccessgrantslocation', 'deleteaccesspoint', 'deleteaccesspointforobjectlambda', 'deleteaccesspointpolicy', 'deleteaccesspointpolicyforobjectlambda', 'deletebucket', 'deletebucketmetadatatableconfiguration', 'deletebucketpolicy', 'deletebucketwebsite', 'deletejobtagging', 'deletemultiregionaccesspoint', 'deleteobject', 'deleteobjecttagging', 'deleteobjectversion', 'deleteobjectversiontagging', 'deletestoragelensconfiguration', 'deletestoragelensconfigurationtagging', 'deletestoragelensgroup', 'describejob', 'describemultiregionaccesspointoperation', 'dissociateaccessgrantsidentitycenter', 'getaccelerateconfiguration', 'getaccessgrant', 'getaccessgrantsinstance', 'getaccessgrantsinstanceforprefix', 'getaccessgrantsinstanceresourcepolicy', 'getaccessgrantslocation', 'getaccesspoint', 'getaccesspointconfigurationforobjectlambda', 'getaccesspointforobjectlambda', 'getaccesspointpolicy', 'getaccesspointpolicyforobjectlambda', 'getaccesspointpolicystatus', 'getaccesspointpolicystatusforobjectlambda', 'getaccountpublicaccessblock', 'getanalyticsconfiguration', 'getbucketacl', 'getbucketcors', 'getbucketlocation', 'getbucketlogging', 'getbucketmetadatatableconfiguration', 'getbucketnotification', 'getbucketobjectlockconfiguration', 'getbucketownershipcontrols', 'getbucketpolicy', 'getbucketpolicystatus', 'getbucketpublicaccessblock', 'getbucketrequestpayment', 'getbuckettagging', 'getbucketversioning', 'getbucketwebsite', 'getdataaccess', 'getencryptionconfiguration', 'getintelligenttieringconfiguration', 'getinventoryconfiguration', 'getjobtagging', 'getlifecycleconfiguration', 'getmetricsconfiguration', 'getmultiregionaccesspoint', 'getmultiregionaccesspointpolicy', 'getmultiregionaccesspointpolicystatus', 'getmultiregionaccesspointroutes', 'getobject', 'getobjectacl', 'getobjectattributes', 'getobjectlegalhold', 'getobjectretention', 'getobjecttagging', 'getobjecttorrent', 'getobjectversion', 'getobjectversionacl', 'getobjectversionattributes', 'getobjectversionforreplication', 'getobjectversiontagging', 'getobjectversiontorrent', 'getreplicationconfiguration', 'getstoragelensconfiguration', 'getstoragelensconfigurationtagging', 'getstoragelensdashboard', 'getstoragelensgroup', 'initiatereplication', 'listaccessgrants', 'listaccessgrantsinstances', 'listaccessgrantslocations', 'listaccesspoints', 'listaccesspointsforobjectlambda', 'listallmybuckets', 'listbucket', 'listbucketmultipartuploads', 'listbucketversions', 'listcalleraccessgrants', 'listjobs', 'listmultipartuploadparts', 'listmultiregionaccesspoints', 'liststoragelensconfigurations', 'liststoragelensgroups', 'listtagsforresource', 'objectowneroverridetobucketowner', 'pausereplication', 'putaccelerateconfiguration', 'putaccessgrantsinstanceresourcepolicy', 'putaccesspointconfigurationforobjectlambda', 'putaccesspointpolicy', 'putaccesspointpolicyforobjectlambda', 'putaccesspointpublicaccessblock', 'putaccountpublicaccessblock', 'putanalyticsconfiguration', 'putbucketacl', 'putbucketcors', 'putbucketlogging', 'putbucketnotification', 'putbucketobjectlockconfiguration', 'putbucketownershipcontrols', 'putbucketpolicy', 'putbucketpublicaccessblock', 'putbucketrequestpayment', 'putbuckettagging', 'putbucketversioning', 'putbucketwebsite', 'putencryptionconfiguration', 'putintelligenttieringconfiguration', 'putinventoryconfiguration', 'putjobtagging', 'putlifecycleconfiguration', 'putmetricsconfiguration', 'putmultiregionaccesspointpolicy', 'putobject', 'putobjectacl', 'putobjectlegalhold', 'putobjectretention', 'putobjecttagging', 'putobjectversionacl', 'putobjectversiontagging', 'putreplicationconfiguration', 'putstoragelensconfiguration', 'putstoragelensconfigurationtagging', 'replicatedelete', 'replicateobject', 'replicatetags', 'restoreobject', 'submitmultiregionaccesspointroutes', 'tagresource', 'untagresource', 'updateaccessgrantslocation', 'updatejobpriority', 'updatejobstatus', 'updatestoragelensgroup']
```

### Expected behavior

Ideally cfn-lint should acknowledge this command and not be showing a warning.

### Reproduction template

Example policy:
```yaml
  ExamplePolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties:
      Description: Example policy
      Groups:
        - ExampleGroupName
      ManagedPolicyName: ExamplePolicyName
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
        - Sid: VerifyAccess
          Effect: Allow
          Action:
          - s3:HeadBucket
          Resource: "*"
```

**Comment by kddejong** [comment]
I believe this is accurate from [docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html)

`General purpose bucket permissions - To use this operation, you must have permissions to perform the s3:ListBucket action. The bucket owner has this permission by default and can grant this permission to others. For more information about permissions, see [Managing access permissions to your Amazon S3 resources](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-access-control.html) in the Amazon S3 User Guide.
`

**Comment by kddejong** [comment]
I can reopen this as needed but I believe this to be the accurate based on the docs.
